internal/tools/flowser.go

```diff
@@ -0,0 +1,122 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package tools
+
+import (
+	"errors"
+	"fmt"
+	"github.com/onflow/flow-cli/internal/command"
+	"github.com/onflow/flow-cli/internal/settings"
+	"github.com/onflow/flow-cli/pkg/flowkit"
+	"github.com/onflow/flow-cli/pkg/flowkit/output"
+	"github.com/onflow/flow-cli/pkg/flowkit/services"
+	"github.com/onflowser/flowser/pkg/flowser"
+	"os"
+	"runtime"
+
+	"github.com/spf13/cobra"
+)
+
+const flowserPath = "flowserPath"
+
+type FlagsFlowser struct{}
+
+var flowserFlags = FlagsWallet{}
+
+var Flowser = &command.Command{
+	Cmd: &cobra.Command{
+		Use:     "flowser",
+		Short:   "Starts Flowser explorer",
+		Example: "flow flowser",
+		Args:    cobra.ExactArgs(0),
+	},
+	Flags: &flowserFlags,
+	RunS:  runFlowser,
+}
+
+func runFlowser(
+	_ []string,
+	_ flowkit.ReaderWriter,
+	_ command.GlobalFlags,
+	_ *services.Services,
+	state *flowkit.State,
+) (command.Result, error) {
+	if runtime.GOOS != settings.Windows && runtime.GOOS != settings.Darwin {
+		fmt.Println("If you want Flowser to be supported on Linux please vote here: https://github.com/onflowser/flowser/discussions/142")
+		return nil, errors.New("OS not supported, only supporting Windows and Mac OS")
+	}
+
+	flowser := flowser.New()
+
+	installPath, err := settings.GetString(flowserPath)
+	if err != nil {
+		return nil, fmt.Errorf("failure reading setting: %w", err)
+	}
+
+	if !flowser.Installed(installPath) {
+		err := installFlowser(flowser, installPath)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	projectPath, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+
+	// check if current directory is existing flow project if not then don't pass project path to Flowser, so user can choose a project
+	_, err = state.ReadFile("foo")
```

Comment:
"foo" just looks weird, "dummyPath" or something more descriptive would be more intuitive.

Reply:
ohh sorry, that's not supposed to be there... still doing some testing, my bad. Thanks for catching